Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2005-01-30
Initial Package Version: 1.0.2
Upstream Status: Not submitted
Origin: None
Description: Use gcc -fpie, -fpic, -fstack-protector-all, and ld -pie,
-z relro, -z now.

See:
http://www.linuxfromscratch.org/~robert/Homogenic/

diff -Naur bzip2-1.0.2.orig/Makefile bzip2-1.0.2/Makefile
--- bzip2-1.0.2.orig/Makefile	2005-01-30 07:35:17.113515771 +0000
+++ bzip2-1.0.2/Makefile	2005-01-30 07:44:23.411010119 +0000
@@ -5,11 +5,14 @@
 CC=gcc
 AR=ar
 RANLIB=ranlib
-LDFLAGS=
+LDFLAGS= -pie -z relro -z now
 
 # Suitably paranoid flags to avoid bugs in gcc-2.7
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -fomit-frame-pointer -fno-strength-reduce $(BIGFILES)
+CFLAGS=-Wall -Winline -O2 -fomit-frame-pointer -fno-strength-reduce \
+-pie -fPIE -fstack-protector-all $(BIGFILES)
+LIBCFLAGS=-Wall -Winline -O2 -fomit-frame-pointer -fno-strength-reduce \
+-fPIC -fstack-protector-all $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
 PREFIX=/usr
@@ -104,19 +107,19 @@
 
 blocksort.o: blocksort.c
 	@cat words0
-	$(CC) $(CFLAGS) -c blocksort.c
+	$(CC) $(LIBCFLAGS) -c blocksort.c
 huffman.o: huffman.c
-	$(CC) $(CFLAGS) -c huffman.c
+	$(CC) $(LIBCFLAGS) -c huffman.c
 crctable.o: crctable.c
-	$(CC) $(CFLAGS) -c crctable.c
+	$(CC) $(LIBCFLAGS) -c crctable.c
 randtable.o: randtable.c
-	$(CC) $(CFLAGS) -c randtable.c
+	$(CC) $(LIBCFLAGS) -c randtable.c
 compress.o: compress.c
-	$(CC) $(CFLAGS) -c compress.c
+	$(CC) $(LIBCFLAGS) -c compress.c
 decompress.o: decompress.c
-	$(CC) $(CFLAGS) -c decompress.c
+	$(CC) $(LIBCFLAGS) -c decompress.c
 bzlib.o: bzlib.c
-	$(CC) $(CFLAGS) -c bzlib.c
+	$(CC) $(LIBCFLAGS) -c bzlib.c
 bzip2.o: bzip2.c
 	$(CC) $(CFLAGS) -c bzip2.c
 bzip2recover.o: bzip2recover.c
diff -Naur bzip2-1.0.2.orig/Makefile-libbz2_so bzip2-1.0.2/Makefile-libbz2_so
--- bzip2-1.0.2.orig/Makefile-libbz2_so	2005-01-30 07:35:17.138508749 +0000
+++ bzip2-1.0.2/Makefile-libbz2_so	2005-01-30 07:44:07.143581514 +0000
@@ -9,7 +9,10 @@
 SHELL=/bin/sh
 CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -fomit-frame-pointer -fno-strength-reduce $(BIGFILES)
+LIBCFLAGS=-fPIC -Wall -Winline -O2 -fomit-frame-pointer \
+-fno-strength-reduce -fstack-protector-all $(BIGFILES)
+CFLAGS=-Wall -Winline -O2 -fomit-frame-pointer -fno-strength-reduce \
+-pie -fPIE -fstack-protector-all $(BIGFILES)
 
 OBJS= blocksort.o  \
       huffman.o    \
@@ -29,16 +32,16 @@
 	rm -f $(OBJS) bzip2.o libbz2.so.1.0.2 libbz2.so.1.0 bzip2-shared
 
 blocksort.o: blocksort.c
-	$(CC) $(CFLAGS) -c blocksort.c
+	$(CC) $(LIBCFLAGS) -c blocksort.c
 huffman.o: huffman.c
-	$(CC) $(CFLAGS) -c huffman.c
+	$(CC) $(LIBCFLAGS) -c huffman.c
 crctable.o: crctable.c
-	$(CC) $(CFLAGS) -c crctable.c
+	$(CC) $(LIBCFLAGS) -c crctable.c
 randtable.o: randtable.c
-	$(CC) $(CFLAGS) -c randtable.c
+	$(CC) $(LIBCFLAGS) -c randtable.c
 compress.o: compress.c
-	$(CC) $(CFLAGS) -c compress.c
+	$(CC) $(LIBCFLAGS) -c compress.c
 decompress.o: decompress.c
-	$(CC) $(CFLAGS) -c decompress.c
+	$(CC) $(LIBCFLAGS) -c decompress.c
 bzlib.o: bzlib.c
-	$(CC) $(CFLAGS) -c bzlib.c
+	$(CC) $(LIBCFLAGS) -c bzlib.c
